as.c

```diff
@@ -31,12 +38,49 @@ static inline vm_operand make_operand(vm_env *env, const char *data)
         op.type = CONST;
         op.value.id = vm_add_const(env, STR, strdup(data + 1));
         break;
+    default:
+        printf(
+            "Error: please specify operand type for '%s' in the following "
+            "line\n",
+            data);
+        printf("\t%s\n\n", line);
+        printf(
+            "Supported types:\n\t$ (constant integer)\n\t# (temp integer)\n\t^ "
+            "(constant string\n");
+        free(line);
+        exit(-1);
+        break;
     }
     return op;
 }
 
+static inline int make_result(vm_env *env, char *line, const char *data)
+{
+    if (data == NULL || data[0] == ';') {
+        printf("Error: missing result in the following line\n");
+        printf("\t%s\n", line);
+        free(line);
+        exit(-1);
+    }
+
+    switch (data[0]) {
```

Comment:
The reasons are:

 1. the consistency of style with 'make_operand'.
2. by heuristic, there should be more target types than one (although not implemented yet).